wallet/psbt.go

```diff
@@ -24,17 +25,21 @@ import (
 // is created and the index -1 is returned.
 //
 // NOTE: If the packet doesn't contain any inputs, coin selection is performed
-// automatically. If the packet does contain any inputs, it is assumed that full
-// coin selection happened externally and no additional inputs are added. If the
-// specified inputs aren't enough to fund the outputs with the given fee rate,
-// an error is returned.
+// automatically, only selecting inputs from the account based on the given key
+// scope and account number. If a key scope is not specified, then inputs from
+// all accounts may be selected, no matter what key scope they belong to. This
+// is done to handle the default account case, where a user wants to fund a PSBT
+// with inputs regardless of their type (NP2WKH, P2WKH, etc.). If the packet
+// does contain any inputs, it is assumed that full coin selection happened
+// externally and no additional inputs are added. If the specified inputs aren't
+// enough to fund the outputs with the given fee rate, an error is returned.
 //
 // NOTE: A caller of the method should hold the global coin selection lock of
 // the wallet. However, no UTXO specific lock lease is acquired for any of the
 // selected/validated inputs by this method. It is in the caller's
 // responsibility to lock the inputs before handing the partial transaction out.
-func (w *Wallet) FundPsbt(packet *psbt.Packet, account uint32,
-	feeSatPerKB btcutil.Amount) (int32, error) {
+func (w *Wallet) FundPsbt(packet *psbt.Packet, keyScope *waddrmgr.KeyScope,
```

Comment:
Same question here, but seems the comment indicates that it's possible for several scoped key managers to have the same default account of 0, so this additional filtration step is needed?

Reply:
Rephrased the comment. Ideally, the key scope wouldn't be optional (note the pointer type) in order to uniquely identify an account and only spend inputs from it. For the case of funding a transaction from the PoV of lnd however, there are cases where we'd want to fund it with a single call of `FundPsbt` that can spend inputs from the two default key scopes it supports (BIP-0049 and BIP-0084).